### PR TITLE
refactor: Replace LRU Dependency With Custom Implementation

### LINKED ISSUE #1368: Retire LRU dependency
The LRU dependency API surface has changed signficnatly between v6 and v7, is it worth migrating to a new version or should we try to implement similar behavior ourselves?

https://github.com/googleapis/google-auth-library-nodejs/pull/1359

### PR DESCRIPTION
**Pull request by danielbankhead**
Fixes #1368 🦕

### CODE REVIEW

**Comment by danielbankhead** [comment]
`do not merge`: Want to add some tests before moving forward.

File: src/util.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+ */
+export class LRUCache<T> {
+  readonly capacity: number;
+
+  /**
+   * Maps are in order. Thus, the older item is the first item.
+   *
+   * {@link https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map}
+   */
+  #cache = new Map<string, {lastAccessed: number; value: T}>();
```
Comment (aeitzman) [inline src/util.ts:41]: nit: private instead of the # notation, unless I'm not understanding what you are trying to do here.

Comment (danielbankhead) [reply src/util.ts:41]: `#` is JS-private (true private) vs TS’s type-only private (that can be ignored).

Comment (aeitzman) [reply src/util.ts:41]: Is there a reason to use them here in particular? Doesn't look like we use the JS-private anywhere else in the code base, also ts style guide says to stick to the typescript private: go/tsjs-style#private-fields (even though that seems to at least partly be for google3 specific reasons)

Comment (danielbankhead) [reply src/util.ts:41]: In the open source orgs (`google`, `googleapis`, etc.) we mostly have our own rules specified in the [`gts`](https://github.com/google/gts) library (historically the tools to enforce the linting rules weren’t available publicly, iirc). And here I think it has a more accurate intention and allows mangling by bundlers to make smaller packages, while `private` cannot.

**Review by aeitzman — APPROVED** [review]

**Pull request merged**